apps/sms/views/inbox/js/inbox.js

```diff
@@ -914,107 +794,57 @@ var InboxView = {
    */
   appendThread: function inbox_appendThread(thread) {
     var timestamp = +thread.timestamp;
-    var firstThreadInContainer = false;
 
     // We create the DOM element of the thread
-    var node = this.createThread(thread);
-
-    // Update info given a number
-    this.setContact(node);
-
-    // Is there any container already?
-    var threadsContainerID = 'threadsContainer_' +
-                              Utils.getDayDate(timestamp);
-    var threadsContainer = document.getElementById(threadsContainerID);
-    // If there is no container we create & insert it to the DOM
-    if (!threadsContainer) {
-      // We create the wrapper with a 'header' & 'ul'
-      var threadsContainerWrapper =
-        InboxView.createThreadContainer(timestamp);
-      // Update threadsContainer with the new value
-      threadsContainer = threadsContainerWrapper.childNodes[1];
-      // Place our new content in the DOM
-      InboxView.insertThreadContainer(threadsContainerWrapper, timestamp);
-      // We had to create a container, so this will be the first thread in it.
-      firstThreadInContainer = true;
-    }
-
-    // Where have I to place the new thread?
-    var threads = threadsContainer.getElementsByTagName('li');
-    var threadFound = false;
-    for (var i = 0, l = threads.length; i < l; i++) {
-      if (timestamp > threads[i].dataset.time) {
-        threadFound = true;
-        threadsContainer.insertBefore(node, threads[i]);
-        break;
+    return this.createThread(thread).then((data) => {
+      // Async update contact after insert to list
+      this.setContact(data);
```

Comment:
I think we could polish this part if we have list cached. Currently you will see the cached list rendered first, data without contacts will be rendered next, and the data with contacts rendered again at last. Maybe we can skip the data without contact rendering step once the cache existed.